cubi_tk/snappy/kickoff.py

```diff
@@ -19,39 +19,45 @@ def run(
 ) -> typing.Optional[int]:
     logger.info("Try to find SNAPPY pipeline directory...")
     try:
-        path = common.find_snappy_root_dir(args.path or os.getcwd(), common.DEPENDENCIES.keys())
+        path = common.find_snappy_root_dir(args.path or os.getcwd())
     except common.CouldNotFindPipelineRoot:
         return 1
 
-    # TODO: this assumes standard naming which is a limitation...
-    logger.info("Looking for pipeline directories (assuming standard naming)...")
+    logger.info("Looking for pipeline directories (needs to contain snappy config.yaml)...")
     logger.debug("Looking in %s", path)
-    step_set = {name for name in common.DEPENDENCIES if (path / name).exists()}
+    step_set = {}
+    folder_steps = common.get_snappy_step_directories(path)
+    for step_name, step_path in folder_steps.items():
+        dependencies = common.get_workflow_step_dependencies(step_path)
+        step_set[step_name] = dependencies
```

Comment:
Why is there no check to ensure that directories with step names actually exist?
From what I can see this is also not checked by common.get_snappy_step_directories(path).

It seems that from here on `step_set`is used both for the steps that actually need to be run *and* for tracking the dependencies between the steps. I think those two things should be tracked separately, i.e. there should be a 'dependencies' dictionary.